File: 2022/01_calorie_counting/part2.py
with open("input.txt", "r") as file:
    sums = []
    sum = 0
    for line in file:
        if line == "\n":
            sums.append(sum)
            sum = 0
        else:
            sum += int(line.strip())
sums.sort(reverse=True)
print(sums[0] + sums[1] + sums[2])

File: 2022/01_calorie_counting/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer input.Close()

	sums := []int{}
	sum := 0

	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := strings.Fields(scanner.Text())
		if len(line) > 0 {
			n, _ := strconv.Atoi(line[0])
			sum += n
		} else {
			sums = append(sums, sum)
			sum = 0
		}
	}

	sort.Sort(sort.Reverse(sort.IntSlice(sums)))
	fmt.Println(sums[0] + sums[1] + sums[2])

	if err := scanner.Err(); err != nil {
		fmt.Println(err)
	}
}
